testing/unittests/esdltests.cpp

```diff
@@ -1527,9 +1551,540 @@ constexpr const char * result = R"!!(<soap:Envelope xmlns:soap="http://schemas.x
+   <Results>
+    <Result>
+     <Dataset xmlns="urn:hpccsystems:ecl:roxieechopersoninfo:result:roxieechopersoninforesponse" name="RoxieEchoPersonInfoResponse">
+      <Row>
+       <Name>
+        <First>aaa</First>
+        <Last>bbbb</Last>
+        <Aliases>
+         <Alias>a</Alias>
+         <Alias>b</Alias>
+         <Alias>c</Alias>
+        </Aliases>
+       </Name>
+       <Addresses>
+        <Address>
+         <Line1>111</Line1>
+         <Line2>222</Line2>
+         <City>Boca Raton</City>
+         <State>FL</State>
+         <Zip>33487</Zip>
+         <type>ttt</type>
+        </Address>
+       </Addresses>
+      </Row>
+     </Dataset>
+    </Result>
+   </Results>
+  </roxieechopersoninfoResponse>
+ </soap:Body>
+</soap:Envelope>)!!";
+
+        constexpr const char *config1 = R"!!(<config>
+          <Transform>
+            <Param name='testcase' value="transform map"/>
+          </Transform>
+        </config>)!!";
+
+        constexpr const char * result = R"!!(<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
+  <soap:Body>
+    <roxieechopersoninfoResponse xmlns="urn:hpccsystems:ecl:roxieechopersoninfo" sequence="0">
+      <Results>
+        <Result>
+          <Dataset xmlns="urn:hpccsystems:ecl:roxieechopersoninfo:result:roxieechopersoninforesponse" name="RoxieEchoPersonInfoResponse">
+            <Row>
+              <Name>
+                <First>aaa</First>
+                <Last>modified-response-at-service-and-method-plus-echoed-alias-echoA2</Last>
+                <Aliases>
+                  <Alias>a</Alias>
+                  <Alias>b</Alias>
+                  <Alias>c</Alias>
+                </Aliases>
+              </Name>
+              <Addresses>
+                <Address>
+                  <Line1>111</Line1>
+                  <Line2>222</Line2>
+                  <City>Boca Raton</City>
+                  <State>FL</State>
+                  <Zip>33487</Zip>
+                  <type>ttt</type>
+                </Address>
+              </Addresses>
+            </Row>
+          </Dataset>
+        </Result>
+      </Results>
+    </roxieechopersoninfoResponse>
+  </soap:Body>
+  <soap:BRESPSRV2>s22</soap:BRESPSRV2>
+  <soap:BRESPSRV3>s33</soap:BRESPSRV3>
+  <soap:BRESPMTH3>m33</soap:BRESPMTH3>
+</soap:Envelope>)!!";
+
+      Owned<IEspContext> ctx = createEspContext(nullptr);
+      Owned<IEsdlScriptContext> scriptContext = createTestScriptContext(ctx, input, config1);
+
+      bool legacy = false;
+      Owned<IEsdlTransformMethodMap> map = createEsdlTransformMethodMap();
+      map->addMethodTransforms("", serverScripts, legacy);
+      map->addMethodTransforms("mymethod", methodScripts, legacy);
+
+      IEsdlTransformSet *serviceSet = map->queryMethodEntryPoint("", "BackendResponse");
+      IEsdlTransformSet *methodSet = map->queryMethodEntryPoint("mymethod", "BackendResponse");
```

Comment:
Just wondering if it would make sense to combine the service and method collections internally (with service transforms shared instead of duplicated) between methods? It could streamline lookup and use of transform sets by decoupling that code from knowledge about how they're defined.

Reply:
Yes it probably would but I'd have to think about the design.  I doubt I'll get that into this PR.  I may open a JIRA.